sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -246,7 +247,27 @@ case class LoadDataCommand(
     val loadPath =
       if (isLocal) {
         val uri = Utils.resolveURI(path)
-        if (!new File(uri.getPath()).exists()) {
+        val filePath = uri.getPath()
+        val exists = if (filePath.contains("*")) {
+          val fileSystem = FileSystems.getDefault
+          val pathPattern = fileSystem.getPath(filePath)
+          val dir = pathPattern.getParent.toString
+          if (dir.contains("*")) {
```

Comment:
It will result in an `AnalysisException` and that seems like the intended behavior. Only a `*` in the file itself is supported.